### PR TITLE
[FIX] utm, *: allow quick creation of campaigns

### PR DESCRIPTION
**Pull request by schoffeniels**
1 - fix `create_name_field` option of many2one field component
-
The `create_name_field` option of many2one fields in form views does not
work: when setting it, the value entered in the many2one field is copied
in the default `name` field, instead of in the field set, when clicking
on the "Create and Edit" button.

This is because an incorrect prop is used: the many2one component uses a
prop named `createNameField` for that purpose, but passes a prop called
`nameCreateField` to the Many2XAutocomplete component.

This commit renames the `createNameField` prop of the many2one field to
`nameCreateField`, to match the prop used in many2many_tags_field and in
Many2XAutocomplete.

2 - fix utm campaigns quick creation
-
Since the introduction of the required "title" field in the utm_campaign
model in commit [1], quick creation of campaigns from many2ones is not
possible anymore (the slow creation modal is opened when clicking on
"Create"), and the name value entered is not copied into the name field
of the creation form.

This is because the model now has a field "name", and a field "title",
which are both required. The creation using the many2one widget tries to
create a record using the entered value as title and no name, which fails.

This commit makes sure that if a title but no name is given when creating
a new campaign record, the name will be computed using the given title.

It also makes sure that the value entered before clicking on the "create
and edit" button is copied in the form, by adding the "create_name_field"
option on the many2one fields (the form uses the title field, not the
name).

[1]: https://github.com/odoo/odoo/commit/4dbcefb5e5b1878e81fe9be9fe48a785f813334f

Task-3113954

### CODE REVIEW

**Comment by Feyensv** [comment]
Fixes #109530 I suppose ;).

File: addons/utm/models/utm_campaign.py [hunk 1]
```diff
@@ -42,6 +42,8 @@ def create(self, vals_list):
         for vals in vals_list:
             if not vals.get('title') and vals.get('name'):
                 vals['title'] = vals['name']
+            elif vals.get('title') and not vals.get('name'):
+                vals['name'] = vals['title']
```
Comment (awa-odoo) [inline addons/utm/models/utm_campaign.py:46]: @std-odoo isn't that the job of the precompute actually?

I think the issue here is that we force a False value in the "name" key of the dict, meaning the precompute does not do its job properly.

I tried changing it this way:
```py
for vals, new_name in zip(vals_list, new_names):
    if new_name:
        vals['name'] = new_name
```

Seems to do what we want when testing locally:

image

Maybe I'm missing something, this is rather complicated :smile:

File: addons/crm/views/crm_lead_views.xml [hunk 2]
```diff
@@ -273,7 +273,7 @@
                                         <field name="company_id"
                                             groups="base.group_multi_company"
                                             options="{'no_create': True}"/>
-                                        <field name="campaign_id" />
+                                        <field name="campaign_id" options="{'create_name_field': 'title', 'always_reload': True}"/>
```
Comment (awa-odoo) [inline addons/crm/views/crm_lead_views.xml:276]: What is the "always_reload" out of curiosity?

**Review by awa-odoo — COMMENTED** [review]
Hello @schoffeniels 

I'm not sure about the python side of the UTM fix, but let's discuss that with @std-odoo .

I left a couple of light comments on the rest.

Thanks & cheers!

File: addons/web/static/src/views/fields/many2one/many2one_field.js [hunk 3]
```diff
@@ -298,7 +298,7 @@ Many2OneField.extractProps = ({ attrs, field }) => {
         canCreateEdit: canCreate && !noCreateEdit,
         relation: field.relation,
         string: attrs.string || field.string,
-        createNameField: attrs.options.create_name_field,
+        nameCreateField: attrs.options.create_name_field,
```
[the remark this answers is not in the document]
Comment (pparidans) [reply addons/web/static/src/views/fields/many2one/many2one_field.js:301]: Indeed, makes sense in stable even though the name disparity looks strange (but already exist in `Many2ManyTagsField` so...).

File: addons/web/static/tests/views/fields/many2one_field_tests.js [hunk 4]
```diff
@@ -3172,6 +3172,40 @@ QUnit.module("Fields", (hooks) => {
...
+        };
+
+        await makeView({
+            type: "form",
+            resModel: "partner",
+            serverData,
+            arch: `
+                <form>
+                    <sheet>
+                        <field name="trululu" options="{'create_name_field': 'foo'}" />
```
[the remark this answers is not in the document]
Comment (pparidans) [reply addons/web/static/tests/views/fields/many2one_field_tests.js:3190]: Aaaah `serverData`... :roll_eyes:

File: addons/utm/models/utm_campaign.py (hunk 1 above)
Comment (schoffeniels) [reply addons/utm/models/utm_campaign.py]: Right, changed it to your approach, thanks :)

**Comment by awa-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**